src/utils/gitUtils.ts

```diff
@@ -21,10 +21,25 @@ import { getSingleRootFsPath } from './workspaceUtils';
 export type GitWorkspaceState = { repo: Repository | null, dirty: boolean, remoteRepo: ReposGetResponseData | undefined; hasAdminAccess: boolean };
 export type VerifiedGitWorkspaceState = GitWorkspaceState & { repo: Repository };
 
+// use only when using the gitApi
+export async function callWithGitErrorHandling<T>(callback: () => Promise<T>): Promise<T | null> {
+    try {
+        return await callback();
+    } catch (gitError) {
+        // ignore empty commit errors which will happen if a user initializes a blank folder or have changes in a nested git repo
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        if (!/nothing to commit/.test(gitError.stdout)) {
```

Comment:
I totally thought this was backwards at first, and I feel like it can't just be me. I think it's because I expect your comment to match the first case in the `if` - something like this:
```
// ignore empty commit errors ...
if (/nothing to commit/.test(gitError.stdout)) {
    return null;
} else {
    throw new GitError(gitError);
}
```